tsdb/engine/tsm1/engine.go

```diff
@@ -1093,9 +1101,15 @@ func (e *Engine) CreateSnapshot() (string, error) {
 	}
 
 	e.mu.RLock()
-	defer e.mu.RUnlock()
+	path, err := e.FileStore.CreateSnapshot()
+	if err != nil {
+		e.mu.RUnlock()
+		return "", err
+	}
+	e.mu.RUnlock()
```

Comment:
I think the engine should still be locked while the index is snapshotted.  If it's unlocked, it's possible the index and tsm files could be inconsistent if deletes and writes were running.